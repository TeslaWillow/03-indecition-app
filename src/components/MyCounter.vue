<template>
  <section class="mb-3">
    <h3>Counter: {{ counter }}</h3>
    <h3 data-testid="square-label">Square: {{ squareCounter }}</h3>

    <div>
      <button class="btn" @click="counter--">-1</button>
      <button class="btn ml-2" @click="counter++">+1</button>
    </div>
  </section>
</template>

<script lang="ts" setup>
import { useCounter } from '@/composables/useCounter';

interface Props {
  value?: number;
};

const props = defineProps<Props>();

const { counter, squareCounter } = useCounter(props.value);
</script>

<style scoped>
.btn {
  @apply p-3 bg-red-600 rounded hover:bg-red-800 text-white;
}
</style>