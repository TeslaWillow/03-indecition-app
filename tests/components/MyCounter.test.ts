import MyCounter from '@/components/MyCounter.vue';
import { mount } from '@vue/test-utils';
import { describe, expect, test } from 'vitest';

describe('<MyCounter />', () => {
  test('Should match snapshot', () => {
    const wrapper = mount(MyCounter, {
      props: {
        value: 15,
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  test('Renders the counter value correctly', () => {
    const value = 5;
    const wrapper = mount(MyCounter, {
      props: {
        value: value,
      },
    });
    console.log(wrapper.html());
    expect(wrapper.find('h3').text()).toContain(`Counter: ${value}`);
    expect(wrapper.find('[data-testid="square-label"]').text()).toContain(
      `Square: ${value * value}`,
    );
  });

  test('Increments the counter when click on btn add + 1', async () => {
    let value = 5;
    const wrapper = mount(MyCounter, {
      props: {
        value: value,
      },
    });
    const btnIncrement = wrapper.find('button.btn.ml-2');
    await btnIncrement.trigger('click');

    value++;
    expect(wrapper.find('h3').text()).toContain(`Counter: ${value}`);
    expect(wrapper.find('[data-testid="square-label"]').text()).toContain(
      `Square: ${value * value}`,
    );
  });
  test('Decrements the counter when click on btn less - 1', async () => {
    let value = 5;
    const wrapper = mount(MyCounter, {
      props: {
        value: value,
      },
    });
    const btnDecrement = wrapper.find('button.btn');
    await btnDecrement.trigger('click');

    value--;
    expect(wrapper.find('h3').text()).toContain(`Counter: ${value}`);
    expect(wrapper.find('[data-testid="square-label"]').text()).toContain(
      `Square: ${value * value}`,
    );
  });
});
