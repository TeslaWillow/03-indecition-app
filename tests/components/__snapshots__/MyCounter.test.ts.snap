// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<MyCounter /> > Should match snapshot 1`] = `
"<section data-v-0352012b="" class="mb-3">
  <h3 data-v-0352012b="">Counter: 15</h3>
  <h3 data-v-0352012b="" data-testid="square-label">Square: 225</h3>
  <div data-v-0352012b=""><button data-v-0352012b="" class="btn">-1</button><button data-v-0352012b="" class="btn ml-2">+1</button></div>
</section>"
`;
